src/basis/basis.js

```diff
@@ -106,53 +114,79 @@ export class Basis {
             ]
         }
      */
-    toJSON() {return JSON.parse(JSON.stringify(_.pick(this, ["elements", "coordinates", "units", "cell"])))}
+    toJSON() {
+        return JSON.parse(
+            JSON.stringify(_.pick(this, ["elements", "coordinates", "units", "cell"])),
+        );
+    }
 
     /**
      * Create an identical copy of the class instance.
      * @param {Object} extraContext - Extra context to be passed to the new class instance on creation.
      */
-    clone(extraContext) {return new this.constructor(Object.assign({}, this.toJSON(), extraContext))}
+    clone(extraContext) {
+        return new this.constructor({ ...this.toJSON(), ...extraContext });
+    }
 
-    getElementByIndex(idx) {return this._elements.getArrayElementByIndex(idx)}
+    getElementByIndex(idx) {
+        return this._elements.getArrayElementByIndex(idx);
+    }
 
-    getCoordinateByIndex(idx) {return this._coordinates.getArrayElementByIndex(idx)}
+    getCoordinateByIndex(idx) {
+        return this._coordinates.getArrayElementByIndex(idx);
+    }
 
-    get elements() {return this._elements.toJSON()}
+    get elements() {
+        return this._elements.toJSON();
+    }
 
-    get elementsArray() {return this._elements.array}
+    get elementsArray() {
```

Comment:
Should we keep the one-line functions?

Reply:
Prettier abhors one-liners but eslint allows it through a `brace-style` rule option. In order for prettier to not clash with eslint (with the rule set), we need to individually ignore files containing one-liners. IMO the use of such an "opinionated" formatter is only beneficial if we release control of what we deem as "stylish" in favor of the ruthless convention.